src/frontend/mame/ui/utils.h

```diff
@@ -260,7 +260,7 @@ bool input_character(std::string &buffer, unicode_char unichar, F &&filter)
 			result = true;
 		}
 	}
-	else if ((unichar >= ' ') && filter(unichar))
+	else if ((unichar >= ' ') && (size == ~0 || buffer.size() < size) && filter(unichar))
```

Comment:
Sorry I'm stumped.

I'm assuming you're not referring to how in the old code, the template parameter was the size of the buffer (including the trailing null), whereas the size limitation in the new code is the size of the string not including the trailing null?  That would be why I'm passing 3 for the size in the new code, whereas the template parameter was 4 in the old code (because it took the size of the buffer)